components/tikv_util/src/future.rs

```diff
@@ -60,61 +41,24 @@ where
     (callback, future)
 }
 
-/// A shortcut for `f1.and_then(|()| f2.flatten())`. Note that
-/// the expression is just a simplified version as f2's Error
-/// type may not be easy handled via combinators.
-pub struct AndThenWith<F1, F2>
-where
-    F2: Future,
-    F2::Item: IntoFuture,
-{
-    f1: Option<F1>,
-    f2: Either<F2, <F2::Item as IntoFuture>::Future>,
-}
-
-impl<F1, F2> AndThenWith<F1, F2>
-where
-    F2: Future,
-    F2::Item: IntoFuture,
-{
-    #[inline]
-    pub fn new(f1: F1, f2: F2) -> AndThenWith<F1, F2> {
-        AndThenWith {
-            f1: Some(f1),
-            f2: Either::Left(f2),
-        }
-    }
-}
-
-impl<E, F2> Future for AndThenWith<Result<(), E>, F2>
+/// Create a stream proxy with buffer representing the remote stream. The returned task
+/// will receive messages from the remote stream as much as possible.
+pub fn create_stream_with_buffer<T, ST>(
```

Comment:
I doesn't seem to be used anywhere.

Reply:
Yeah, but it will be used in future work. Like here: https://github.com/tikv/tikv/pull/8447/files#diff-ff1ef1465cb21542cabd28608a3e7e49R134-R138. It's designed to replace `mpsc::spawn` in futures 0.1.